src/components/projects/contracts/ContractsForm.tsx

```diff
@@ -0,0 +1,264 @@
+                        className="grid md:grid-cols-3 grid-cols-1 gap-2"
+                      >
+                        {HasDeployerKeysOption.options.map((option) => (
+                          <FormItem key={option}>
+                            <FormLabel className="flex-1 min-w-6 basis-0 p-4 text-sm font-medium flex items-center gap-3 border rounded-sm">
+                              <FormControl>
+                                <RadioGroupItem value={option} />
+                              </FormControl>
+                              {option}
+                            </FormLabel>
+                          </FormItem>
+                        ))}
+                      </RadioGroup>
+                    </FormControl>
+                    <FormMessage />
+                  </FormItem>
+                )}
+              />
+            </div>
+          </div>
+
+          {formValues.hasDeployerKeys !== "No" && (
+            <div className="flex flex-col gap-6">
+              <div className="flex flex-col gap-3">
+                <h3>Onchain verification</h3>
+                <div className="text-text-secondary">
+                  First verify one contract, then you&apos;ll be able to add
+                  more. Additional contracts with the same deployer address will
+                  be automatically verified.
+                </div>
+              </div>
+              {contractsFields.map((field, index) => (
+                <ContractForm key={field.id} form={form} index={index} />
+              ))}
+            </div>
+          )}
+
+          {formValues.hasDeployerKeys !== "Yes" && (
+            <div className="flex flex-col gap-6">
+              <h3>Add this project to Open Source Observer</h3>
+              <div className="text-text-secondary">
+                It is highly encouraged that projects verify contracts onchain.
+                However, if you&apos;ve lost your deployer keys, you can
+                complete this step by adding your project to{" "}
+                <span className="font-medium">Open Source Observer.</span>
+              </div>
+              <div className="text-text-secondary">
+                Follow{" "}
+                <Link className="font-medium" href="#">
+                  these instructions
+                </Link>{" "}
+                for adding your project. Make sure that your project has been
+                added before the Retro Funding submission deadline.
+              </div>
+              <Button
+                className="p-0 self-start"
+                type="button"
+                variant="secondary"
+              >
+                <Link
+                  className="flex items-center gap-2.5 w-full h-full py-2 px-3 font-medium"
+                  href="#"
+                >
+                  View instructions{" "}
+                  <Image
+                    src="/assets/icons/arrow-up-right.svg"
```

Comment:
could probably use a lucide icon for this but nbd either way

Reply:
ah true